### DISCUSSION TITLE
How to update legacy version?

### DISCUSSION THREAD

**Discussion by the-hotmann** [shlinkio/discussions#65] [Help wanted] [answered]
* Shlink Version: Server: v2.10.0
* PHP Version: 8.1.6
* How do you serve Shlink: Self-hosted Apache
* Database engine used: MariaDB

#### Summary
Previously I updated Shlink server with `php bin/update` but this is not possible anymore since the file `bin/update` is missing. in the docs and the readme the "Update" part apparently disappeared. So I would like to ask for an proper way to update to the newest Shlink server version :)

Docker is no problem for me, but a support for native `.pem` files (for SSL) would be good, otherwise HTTPS (SSL) would need to be added with a reverseProxy which I could do, but would not appreciate as a dependency.

**Comment by acelaya** [comment, marked as answer]
All the breaking changes introduced in major versions are always documented in the UPGRADE.md file. It's a good idea to review that when doing a major version upgrade.

This section explains how the new installer tool entry points work https://github.com/shlinkio/shlink/blob/develop/UPGRADE.md#changes-in-cli

Regarding the documentation, it's always been here https://shlink.io/documentation/update-your-instance/#updating-a-self-hosted-instance. However, I just realized there's still some references to the old scripts there, that I will fix ASAP.

**Comment by the-hotmann** [comment]
After I read it another time I indeed saw the problem. The command
`vendor/bin/shlink-installer update`
indeed was right, but the problemn was, it is not shell script and therefore I indeed did not execute it as a PHP file like
`php vendor/bin/shlink-installer update`
but instead as bash/sh script.

Maybe you should make that clear by adding this to the command. Indeed by checking the content of the script I should have known, but  I did not check the content ... my fault. Sorry.

So I at least like to ask for a more clear statement, that this script `vendor/bin/shlink-installer` is a PHP script and shall be executed like this (or something else)
`php vendor/bin/shlink-installer update` instead of `vendor/bin/shlink-installer update` as the later command would execute it as a shell script.


Thanks!
